storage/namespace.go

```diff
@@ -537,8 +582,10 @@ func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
 		if shard == nil {
 			continue
 		}
-		if shard.FlushState(blockStart).Status == fileOpInProgress {
-			return false
+		for _, blockStart := range blockStarts {
```

Comment:
The original code seems to be use this check to ensure only a single flush is happening at a time. The flush manager is already [doing that](https://github.com/m3db/m3db/blob/c93ecb1dc6806088f616b5ff8cf76960e9acba90/storage/flush.go#L82-L98). I'm fine with removing if you are - @robskillington @xichen2020 ?